pkg/util/validation/limits_test.go

```diff
@@ -586,6 +587,193 @@ testuser:
 	}
 }
 
+func TestRulerMaxRulesPerRuleGroupLimits(t *testing.T) {
+	tc := []struct {
+		name      string
+		inputYAML string
+		expected  struct {
```

Comment:
nit: can we use `expectedLimit`, `expectedNamespace`, and avoid repeating struct definition in every test?

Reply:
Done in https://github.com/grafana/mimir/pull/8378/commits/426289646b3ba7479d456e82f20b74c3432a9bd2